envoy/go.mod

```diff
@@ -2,8 +2,12 @@ module github.com/envoyproxy/go-control-plane/envoy
 
 go 1.22
 
+// Used to resolve import issues related to go-control-plane package split (https://github.com/envoyproxy/go-control-plane/issues/1074)
+replace github.com/envoyproxy/go-control-plane@v0.13.3 => ../
```

Comment:
nit: unless you need this for your project replace statements don't do anything for downstream users. So if you don't need this it can be removed.

Reply:
Thank you for the review. I took this from https://go.dev/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository, where they explicitly state this replace as a way to avoid the "ambiguous" import.
I do not have enough understanding of go.mod resolution logic to know if this actually helps or not

Reply:
I believe the section you are referring to is:

```
cd path-to/github.com/my-repo/mig
go mod edit -replace github.com/my-repo@v1.3.0=../
go test ./...
go mod edit -dropreplace github.com/my-repo@v1.3.0
```

If so this is meant to be a temporary test, as you can see be the drop command after. This is because it assumes the module is not yet published that you are referencing. In this case though the version where this module has been carved out already exists. 

This causes no harm either way though, just thought I would share :)

Reply:
Thanks for the update. I think I get it better now, and they cut tag out of the intermediate commit with the replace. In order to keep it simpler to track I'll do this in two PRs. I'll create a new one to drop the replace